### DISCUSSION TITLE
Converting a Second Coordinator to a Zigbee Router

### DISCUSSION THREAD

**Discussion by DDS-WorkFlow** [tube0013/tube_gateways#202] [Q&A] [answered]
Hi everyone,

I’m relatively new to Zigbee and recently encountered some challenges while trying to expand my Zigbee network. I’m hoping to get some guidance.
Background:
I initially purchased a TubeZB CC2652P7-based coordinator for my Zigbee network, which worked great. Wanting to extend the network further, I made the mistake of buying a second TubeZB CC2652P7 coordinator, thinking it could also function as an extender/repeater/router. I only realized my mistake when I tried to configure it in Zigbee2MQTT and found out that Home Assistant does not allow two coordinators in the same network.

What I’ve Done So Far:
Flashed the Router Firmware:

I used the firmware tubeszb_CC1352P2_CC2652P_launchpad_router_20221102.hex.zip, believing it would configure my second coordinator to function as a router.
The flashing process completed successfully, with no errors.

Checked Device Status:

I attempted to confirm that the device was in router mode, but the TubeZB web interface does not provide an option to confirm this status. The UI remains the same as it was before flashing, without specific indicators for router mode.

Configured Zigbee2MQTT:

Ensured my configuration.yaml was set up correctly, enabling permit_join to allow new devices to join the network.
Restarted Zigbee2MQTT and monitored the logs, but the new router was not being recognized.

Confirmed Wireless Range Requirements:

I learned that, despite the PoE connection, the router needs to be within wireless range of the coordinator or another router. I adjusted its placement accordingly but still faced issues with it appearing in the network map.
Issues Encountered:
The new router is not being recognized by Zigbee2MQTT, although ZHA (Zigbee Home Automation) and ESPHome are picking it up, however I think it's still registering in them as a coordinator.
There were no errors during the flashing process, and the device’s web interface does not show any specific router mode indicators.
Despite enabling permit_join and placing the router within range, it does not appear in the Zigbee2MQTT logs or network map.

Additional Considerations:
I’m not even sure if converting a coordinator to a router is feasible with this hardware and firmware setup. If it turns out this approach doesn’t work, I am willing to purchase the TubeZB repeater once it is back in stock. However, I wanted to avoid wasting the additional coordinator I bought.

Request for Help:
Could anyone provide insights or suggestions on what might be going wrong? Is there a specific step I might have missed in the configuration or flashing process? Any advice on how to ensure Zigbee2MQTT recognizes the new router would be greatly appreciated.

Additional Information:
I am using the latest version of Zigbee2MQTT.
I have ensured there are no IP conflicts or network issues.
I have followed the available documentation and community forums but haven't found a solution yet.
Thank you in advance for any help or guidance you can provide!

**Comment by tube0013** [comment, marked as answer]
Hi,

TubesZB is just me..

The 1 coordinator per zigbee network is a limitation of zigbee not z2m or homeassistant.

the router FW your flashed is only for the cc2652P2 it will not work on the P7. The Z-Stack Repo whose FW the router FW is based does not yet have a router version for the p7. Mercenaruss who has the zigstar has a router FW on github but I've never tested it and do not know how well it works: https://github.com/mercenaruss/zigstar_gateways/tree/main/files/Firmware

I'd give this FW a shot and see how it goes.

**Comment by DDS-WorkFlow** [comment]
Thanks for pointing me in the right direction!

I ended up using the zr_genericapp_LP_CC1352P7_4_tirtos7_ticlang.hex firmware on the TubeZB CC2652P7. After configuring Zigbee2MQTT and some troubleshooting, the router finally showed up in the network and is working as expected.

Appreciate your help!
